### PR TITLE
Simplified contribution guidelines

### PR DESCRIPTION
**Pull request by mmattioli**
Simply moved `CONTRIBUTING.md` into `.github` for the sake of simplification and organization

### CODE REVIEW

**Comment by xwu** [comment]
A link to CONTRIBUTING.md is presented at the top of every pull request being created. Like README.md, it's a magical part of GitHub.

See: https://github.com/blog/1184-contributing-guidelines

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -9,5 +10,14 @@ Before merging this pull request, you must run the Swift continuous integration
 For information about triggering CI builds via @swift-ci, see:
 https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci
 
+By submitting a pull request, you represent that you have
+the right to license your contribution to Apple and the community, and agree by submitting the patch
```
Comment (xwu) [inline .github/PULL_REQUEST_TEMPLATE.md]: Is there a reason these lines are wrapped so raggedly?

**Comment by mmattioli** [comment]
@xwu I'm more familiar with the [Issue and Pull Request template feature](https://github.com/blog/2111-issue-and-pull-request-templates) that was more recently announced; in spirit of that feature and for the sake of organization, I updated this PR with alternative approach.

**Comment by xwu** [comment]
It's easier to find CONTRIBUTING.md when it's at the root of the repo, don't you think?

**Comment by mmattioli** [comment]
The purpose of its existence is to make its content accessible to users; if we've already established that a GitHub feature makes its content easily accessible whenever someone opens a PR or Issue and we've also established that this file can be placed in either the root of the repo _or_ in the `.github` directory then why not place it in the `.github` directory with the other file(s) (i.e. `PULL_REQUEST_TEMPLATE.md`) which take advantage of the aforementioned GitHub features? `PULL_REQUEST_TEMPLATE.md` could also be placed in the root of the repository and provide the same functionality it currently provides but neither that file or `CONTRIBUTING.md` are specifically mentioned or formally referenced in any documentation therefore making their purpose for existence a means to display their contents to users through the aforementioned GitHub features.

**Comment by xwu** [comment]
The custom of placing CONTRIBUTING.md at the root of the repo pre-dates the GitHub feature, and there are those who would look for it in that location. For instance, a person who has checked out a local working copy may wish to consult that document while browsing through the folder on their file system, independent of the GitHub user interface, just as it may be the case for README.md. By contrast, the PR template is not meant to be useful independent of GitHub, which is why it is in a hidden folder.

**Comment by mmattioli** [comment]
Your logic is based on functionality that is independent of GitHub but the contents of `CONTRIBUTING.md` specifically reference and instruct users on how to make contributions __using GitHub__; furthermore, the contents of the file are specific to how one should go about opening a pull request which is a feature of GitHub.

@practicalswift care to weigh in on this?

**Comment by jrose-apple** [comment]
I'm with @xwu on this. It's also a bit of a process and presentation thing, so @tkremenek and @DougGregor might be the best people to make the final call.

**Comment by xwu** [comment]
Not all open-source projects on GitHub accept changes via GitHub. (Some, for instance, are mirrors.) The contents of CONTRIBUTING.md answer certain questions for a prospective contributor (in the case of Swift, via a link to a more extensive page of guidelines):

* What sorts of contributions are welcomed?
* How can it be contributed?
* What are the legal conditions surrounding contributed code?

This is important information to have before someone begins tackling a bug and does not necessarily involve GitHub--it just so happens that Swift accepts contributions via GitHub. By contrast, the PR template addresses other questions. Namely, now that you've already made some changes you would like to contribute and are about to open a PR:

* What is the content of those changes?
* What bugs, if any, have you resolved?
* How do you go about testing and merging those changes upstream?

But of course, it's up to core team members to make the call. Just putting in my two cents.

**Comment by tkremenek** [comment]
I respect the motivation behind this change, but I agree with @xwu's arguments about keeping this at the root.  I think it accomplishes two things (which @xwu outlined):

* It makes the contribution guidelines easily discoverable on the local file system, and elevates their importance.
* There is a separation of concerns (IMO) between contribution guidelines to the project and the hosting infrastructure, in this case GitHub.  While the guidelines may naturally reference that infrastructure, the guidelines can encompass other things like the points @xwu outlined, such as legal considerations, etc.

The second point is a bit subtle—we would have contribution guidelines *anyway* even if the project was not hosted on GitHub, and we would naturally put them at the top as a peer of README.

**Pull request closed without merging**